projects/website/markdown/posts/2022-03-28-tuist-3.0/post.mdx

```diff
@@ -0,0 +1,50 @@
+
+Plugins have been in tuist for some time now but it is now possible to use **third-party dependencies in plugin [tasks](https://docs.tuist.io/plugins/creating-plugins/#tasks)**. This unlocks a lot of possibilities. And when users define a task in their `Config.swift`, it's integrated right inside the tuist CLI. For example, we have created a [new tuist plugin](https://github.com/tuist/tuist-plugin-lint) for linting source code where we have integrated the SwiftLint package to actually do the linting. When you integrate the plugin in your project, you can just run `tuist lint` and it will trigger that plugin 🤯  How _cool_ is that?
+
+But there's more. `tuist lint` also needs to know which files it should lint. For such a use-case, you can now use a completely new framework called [ProjectAutomation](https://docs.tuist.io/guides/task/#projectautomation). This framework gives **tasks access to the graph** and a plugin like `tuist lint` can now query the list of sources:
+
+```swift
+import ProjectAutomation
+let allSources = Tuist.graph().projects.values.flatMap(\.targets).map(\.sources)
+```
+
+Having an access to the graph is extremely powerful and can be used for _so many_ things - and we can't wait to learn what you build.
+
+## Dependencies.swift
+
+We have all had our share of pain of using the SPM integration in Xcode. And more importantly, you can only integrate the SPM targets as sources - so when you inevitably clean your project, you need to _rebuild_ all of the dependencies again. That's why still a lot of users stick to using Carthage. You always integrate the targets as frameworks, never having to worry about building that code multiple times. Additionally, your Xcode projects is snappier because it only links binaries. The drawback of Carthage has always been you as a developer have to do a lot of manual setup. And when the time comes and you have to debug your dependency, you need to go back to SPM or Cocoapods.
+
+With [`Dependencies.swift`](https://docs.tuist.io/manifests/dependencies/), you declare your dependencies easily inside a dedicated Swift file instead of relying on Xcode UI. Your dependencies are resolved via a simple command `tuist fetch`. And then when you generate your project, tuist prebuilds all the dependencies, so you get a lean Xcode project which you can clean without worrying about ever having to rebuild your dependencies. And if you still want to debug a dependency, simply specify it in your `tuist generate` call.
```

Comment:
Shouldn't we point to the new DocC documentation?

Reply:
I wouldn't do so for now. And with docc, I think we should keep the high-level explainer of the manifests and only use docc for the description of all the available properties.

Reply:
I think the direction we should take (and AFAIU we are taking [here](https://github.com/tuist/tuist/pull/4285)) is to have the documentations of the manifests in code, because otherwise it's easy for them to diverge.
The missing parts (e.g. the example we have at the beginning of https://docs.tuist.io/manifests/dependencies/) can be added to DocC/code documentation as well I guess

Reply:
cc @adellibovi

Reply:
Oops, I was also under the impression/understanding, we would move the overview of the manifest there as well, as it is supported by DocC and usually only a couple of lines and a snippet.

I am attaching few screenshots so you can see how they compare. @fortmarek let me know what do you think.
Note: I attached the ones from Xcode simply because it's easier to generate, on the online version they will look the same but with more screen estate since the Xcode sidebar won't be there.

<img width="1136" alt="Screenshot 2022-03-28 at 18 40 33" src="https://user-images.githubusercontent.com/67916/160446260-6e85c085-737a-4908-b67b-9f023c407545.png">
<img width="1260" alt="Screenshot 2022-03-28 at 18 40 50" src="https://user-images.githubusercontent.com/67916/160446306-0b949d4f-9051-4d98-bbf7-1ac66fcdeee0.png">
<img width="1136" alt="Screenshot 2022-03-28 at 18 40 27" src="https://user-images.githubusercontent.com/67916/160446242-5911475e-4a17-4368-b357-680fb6b8da82.png">
<img width="1260" alt="Screenshot 2022-03-28 at 18 41 10" src="https://user-images.githubusercontent.com/67916/160446368-54ab2150-e88b-42a6-8e93-1fe026af640f.png">

Reply:
I think the issue is that docc is not gonna be inside our documentation - and searching for something like `Dependencies.swift` will not give you any results if you are in docs.tuist.io. That's why I would keep the high-level explainer in the docs.tuist.io and for the full description of the API, delegate to docc.